tests/preprocessing/test_label.py

```diff
@@ -138,7 +121,14 @@ def test_categorical(self, categories, transformed, daskify, ordered):
 
         if daskify != "unknown":
             assert a.dtype_ == cat.dtype
-        np.testing.assert_array_equal(a.classes_, categories)
+
+        if DASK_EXPR_ENABLED and daskify == "unknown" and categories != ["b", "a"]:
+            with pytest.raises(AssertionError, match="Arrays are not equal"):
+                np.testing.assert_array_equal(a.classes_, categories)
+            return
```

Comment:
@phofl 

<details>
<summary>traceback</summary>

```
    def test_categorical(self, categories, transformed, daskify, ordered):
        cat = pd.Series(
            ["a", "b", "a"],
            dtype=pd.api.types.CategoricalDtype(categories=categories, ordered=ordered),
        )
        if daskify:
            cat = dd.from_pandas(cat, npartitions=2)
            transformed = da.from_array(transformed, chunks=(2, 1))
            if daskify == "unknown":
                cat = cat.cat.as_unknown()

        a = dpp.LabelEncoder().fit(cat)

        if daskify != "unknown":
            assert a.dtype_ == cat.dtype

>       np.testing.assert_array_equal(a.classes_, categories)

tests/preprocessing/test_label.py:125:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array(['b', 'a'], dtype=object), ['a', 'b'])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 2 / 2 (100%)
E            x: array(['b', 'a'], dtype=object)
E            y: array(['a', 'b'], dtype='<U1')

../../mambaforge/envs/dask-ml/lib/python3.10/contextlib.py:79: AssertionError
```

</details>

Reply:
This could be legit, could you look into why the order is not correct?